analysis/taint/dataflow_visitor.go

```diff
@@ -155,6 +155,15 @@ func (v *Visitor) Visit(s *df.AnalyzerState, source df.NodeWithTrace) {
 			} else {
 				if v.taints.addNewPathCandidate(NewFlowNode(v.currentSource), NewFlowNode(cur.NodeWithTrace)) {
 					reportTaintFlow(s, v.currentSource, cur)
+					s.Report.AddEntry(s.Logger, s.Config, config.ReportDesc{
+						Tool:     "taint",
+						Tag:      v.taintSpec.Tag,
+						Severity: v.taintSpec.Severity,
+						Content:  report(s, v.currentSource, cur, v.taintSpec),
+					})
+					if v.taintSpec.Severity == config.Critical {
+						panic(fmt.Sprintf("taint flows to critical location (problem tag %s)", v.taintSpec.Tag))
```

Comment:
We'll have to adjust, and the best way to handle this would probably to have another process listen to the results. We also don't want the user to wait 2 hours if a critical problem was encountered in the first minutes; that critical problem should be resolved before proceeding.